### ISSUE TITLE
Dlib-Facenet face detection comparison

### ISSUE THREAD

**Issue by RiccardoForniz** [davidsandberg/facenet#1023]
Hello everyone,
i really would like to know if i made any mistake with face detection through facenet  because 
in my project it has some problems with face detection when in the picture the face is rotated or if it is partially occluded, but on the other side using Dlib there is no problem with face detection on the same images, then, reading the David's description Facenet should be more accurate than Dlib.
Surely i am missing something in my project.
I'have uploaded the processed images, the firsts are processed through Dlib, the seconds through Facenet, attached you can find also the original images.

![img_dlib1558431469 9018571](https://user-images.githubusercontent.com/18027558/58086365-caec5780-7bbe-11e9-8130-6600d2f3772e.jpg)
![img_dlib1558431790 7402456](https://user-images.githubusercontent.com/18027558/58086366-cb84ee00-7bbe-11e9-8b9e-62f125de4a02.jpg)
![img_dlib1558431312 2799883](https://user-images.githubusercontent.com/18027558/58086367-cb84ee00-7bbe-11e9-8635-4256d7e38341.jpg)
![img_facenet1558431556 1939394](https://user-images.githubusercontent.com/18027558/58086394-d9d30a00-7bbe-11e9-8705-07e642cafb33.jpg)
![img_facenet1558431679 0756643](https://user-images.githubusercontent.com/18027558/58086396-d9d30a00-7bbe-11e9-9c6d-ab91fcabc823.jpg)
![img_facenet1558430394 603888](https://user-images.githubusercontent.com/18027558/58086398-d9d30a00-7bbe-11e9-9b94-0e51874d94ac.jpg)

![occlusion_2](https://user-images.githubusercontent.com/18027558/58086417-e5becc00-7bbe-11e9-884d-78b11f439144.jpg)
![rotation_web](https://user-images.githubusercontent.com/18027558/58086418-e5becc00-7bbe-11e9-8348-d113e02f03a7.png)
![multi_occlusion2](https://user-images.githubusercontent.com/18027558/58086420-e6576280-7bbe-11e9-871d-ef177d340b77.png)

Could you please try with your implementation of Facenet and share with me the results?

These are my configs params 
      minsize = 20  # minimum size of face
        threshold = [0.6, 0.7, 0.7]  # three steps's threshold
        factor = 0.709  # scale factor
        margin = 44
        batch_size = 1000
        image_size = 182
        input_image_size = 160

Thanks guys

**Comment by shehel** [comment]
Facenet uses MTCNN for face detection. If you haven't changed any detection hyperparameters, then it could be that dlib face detection is better than MTCNN or you might need to change hyperparameters for this dataset which I doubt is the case.

**Comment by RiccardoForniz** [comment]
Yes sorry i forget to specify that i use MTCNN for face detection, than is there a way to improve the MTCNN performances?
Or did i make any mistake during the configuration of my project?
Could you confirrm my results with yours imlementation? 
If your results are better then mines could you please share with me your MTCNN Configuration?
Thanks guys

**Comment by freedom1810** [comment]
i think it's because the method you load the image, try load image in brg mode instead rgb, it will return the best accuracy,

**Comment by faybak** [comment]
hello @RiccardoForniz 
did you find solution about it. I would want to know if there is a configuration of MTCNN which improve his performance please

**Comment by RiccardoForniz** [comment]
Hello @faybak, no I don't, in the end I used Dlib CNN as extractor features and alignment.
Ric.

**Comment by RajkiranVeldur** [comment]
Hello, @RiccardoForniz have you tried using Face detector as MTCNN and 68 landmarks of Dlib?  Actually, I'm trying to implement it, but I'm unable to get it to work.

**Comment by rajhlinux** [comment]
I have done some comparison performance between dlib and MTCNN. From my observation, MTCNN is no better than dlib's face detection.

If you want a professional super accurate 100% of the time of face detection, I would recommend "libfacedetection" over MTCNN:

https://github.com/ShiqiYu/libfacedetection

I'm not sure what algo is used in libfacedetection, I think it is a highly modified and advanced form of MTCNN. My tests shows that "libfacedetection" is super accurate and have zero false negatives. Note: I have used many MTCNN algos, none beats the detection accuracy of libfacedetection algo.
